packages/cli/src/lib/setup/setupRepo.ts

```diff
@@ -1,55 +1,90 @@
-'use strict';
-
-/** @class */
-function Repo(options) {
-    const { EXIT_CODES } = require('@iobroker/js-controller-common');
-    const { tools } = require('@iobroker/js-controller-common');
-    const axios = require('axios');
-    const ioPackage = require(`${tools.getControllerDir()}/io-package.json`);
-    const version = ioPackage.common.version;
-    const fs = require('fs');
-
-    const defaultSystemRepo = {
-        common: {
-            name: 'System repositories',
-            dontDelete: true
-        },
-        native: {
-            repositories: {
-                stable: {
-                    link: 'http://download.iobroker.net/sources-dist.json',
-                    json: null
-                },
-                beta: {
-                    link: 'http://download.iobroker.net/sources-dist-latest.json',
-                    json: null
-                }
-            }
-        },
-        _id: 'system.repositories',
-        type: 'config'
-    };
+import { tools, EXIT_CODES } from '@iobroker/js-controller-common';
+import axios from 'axios';
+import fs from 'fs-extra';
+import type { Client as ObjectsRedisClient } from '@iobroker/db-objects-redis';
+import type { Client as StatesRedisClient } from '@iobroker/db-states-redis';
+import path from 'path';
+
+export interface CLIRepoOptions {
+    objects: ObjectsRedisClient;
+    states: StatesRedisClient;
+}
 
-    options = options || {};
+export interface RepoFlags {
+    /** Also list not installed adapters */
+    a?: boolean;
+    all?: boolean;
+    /** Only list updatable adapters */
+    u?: boolean;
+    updatable?: boolean;
+    /** Force update even if hash hasn't changed */
+    f?: boolean;
+    force?: boolean;
+}
 
-    if (!options.objects) {
-        throw new Error('Invalid arguments: objects is missing');
-    }
-    if (!options.states) {
-        throw new Error('Invalid arguments: states is missing');
-    }
+export class Repo {
+    private readonly defaultSystemRepo: Record<string, any>;
```

Comment:
```suggestion
    private readonly defaultSystemRepo: ioBroker.OtherObject;
```
also solves your @ts-expect-error below.